### PR TITLE
Fix CREATE DATABASE … ENCRYPTION syntax

### PR DESCRIPTION
**Pull request by zhenghaoz**
#### Fix CREATE DATABASE … ENCRYPTION syntax

Issue: pingcap/parser#417

#### MySQL Syntax:

```
create_specification:
...
  | DEFAULT ENCRYPTION [=] {'Y' | 'N'}
```

### Bad SQL Case:

```sql
CREATE DATABASE t ENCRYPTION = 'N';
CREATE DATABASE t ENCRYPTION 'N';
CREATE DATABASE t DEFAULT ENCRYPTION = 'N';
CREATE DATABASE t DEFAULT ENCRYPTION 'N';

CREATE DATABASE t ENCRYPTION = 'Y';
CREATE DATABASE t ENCRYPTION 'Y';
CREATE DATABASE t DEFAULT ENCRYPTION = 'Y';
CREATE DATABASE t DEFAULT ENCRYPTION 'Y';
```

### Check List

Tests
- Unit test

### CODE REVIEW

**Comment by zhenghaoz** [comment]
It seems that an error should be raised when the value for `ENCRYPTION` is neither `Y` nor `N`.
```sql
CREATE DATABASE t ENCRYPTION = 'xxx';
```
In MySQL 8.0, the raised error is:
```
ERROR 1525 (HY000): Incorrect argument (should be Y or N) value: 'xxxxx'
```
The error handling seems necessary but I don't know where we could declare an error type so that we can check the value in `parser.y`.

**Comment by kennytm** [comment]
Let's ignore the error when the value is neither N or Y. This could be checked after parsing if needed.

File: parser.y [hunk 1]
```diff
@@ -2429,6 +2434,10 @@ DatabaseOption:
 	{
 		$$ = &ast.DatabaseOption{Tp: ast.DatabaseOptionCollate, Value: $4.(string)}
 	}
+|	DefaultKwdOpt "ENCRYPTION" EqOpt EncryptionOption
```
Comment (kennytm) [inline parser.y]: Let's just use `stringLit` here (and remove `EncryptionOption`). Like `TableOption` we normally don't create a separate rule for every option unless it is necessary.

```suggestion
|	DefaultKwdOpt "ENCRYPTION" EqOpt stringLit
```

**Comment by leoppro** [comment]
@zhenghaoz resolve the conflicts please

**Comment by zhenghaoz** [comment]
@leoppro done

**Pull request merged**